### ISSUE TITLE
Add gas consumption

### ISSUE THREAD

**Issue by danygi79** [wrongisthenewright/ebusd-configuration-ariston-bridgenet#20]
HI,
Is there a way to add the gas consumption of the boiler in addition to the electricity consumption?
I have a hybrid system from Chaffoteaux and I am forced to read that data via cloud integration.

thank you all

**Comment by wrongisthenewright** [comment]
I haven’t been able to find the gas consumption metric in the bus messages. It’s either not present or, more probably, not yet decoded.

**Comment by wrongisthenewright** [comment]
,energymgr,heatpump_calculated_cost

**Comment by wrongisthenewright** [comment]
...

**Comment by wrongisthenewright** [comment]
I'm working on a hint and assumption, that the code 6847 represent in some way the gas consumption or that rather it is proportional to it.
I've found that plotting the grapho of that parameter in HA I can correlate it with boiler usage either in DHW mode or CH mode, furthermore the graph have values that are compatible with a gas consumption related metric.

**Comment by wrongisthenewright** [comment]
I think I've made an interesting discovery!
I was scrolling trhu my boiler tech menu and entered the boiler diagnostic menu, there, among others there is a menu that shows the _current_ boiler power expressed in kW. It's not the nominal power rather the current power output of the boiler expressed in kW.
This is not a direct measurement of the gas consumption but, with a rough estimation, we can convert kW to natgas m3 dividing by 11,06 (or multiplying by 0.0904) taking the standard MJ/m3 conversion ratio.
Other gas (LPG, Buthane) need appropriate conversion ratios.

The parameter is 0x6847, I'll add that at the next release!

**Comment by danygi79** [comment]
Thank you for your work. 

I thought that the indication should be 2, one for domestic water and one for technical water for heating.

I can help you find out my explanations how to do it, I'll try.

**Comment by wrongisthenewright** [comment]
In my system that parameter is used and updated in both cases (dhw and heating) so it’s not possible to know from that parameter alone what is using gas.

If you want to split the computing you need to take into account the operating mode eg. with a template in HA.

**Comment by danygi79** [comment]
hi, couldn't you enter the parameter in the files?

**Comment by wrongisthenewright** [comment]
Yes I’ll do it in the next release. I need to see if it works for a little time more as my boule is doing mostly DHW as heating is done by the HP

**Comment by danygi79** [comment]
ok, you're in the same situation as me, gas used only for DHW and for heating only if external temperatures drop below -2°C, very rarely here.

**Comment by wrongisthenewright** [comment]
Fixed in v2.6

From the few test I made it seems quite good, final judgement will be at the end of the heating season comparing the calculated data from ebusd with the actual consumption from my gas meter. 
Please re-open the issue if you feel the data is incorrect
